src/pybind/mgr/mgr_module.py

```diff
@@ -2,8 +2,11 @@
 
 import logging
 import json
+import netifaces
```

Comment:
i'd suggest avoid introducing yet another dependency. how about:

```python
import socket
# ...
DEFAULT_ADDR = '::' if socket.has_ipv6 else '0.0.0.0'
```
?

Reply:
socket.has_ipv6 always ture even disable ipv6

Reply:
> socket.has_ipv6 always ture even disable ipv6

Have a look here https://www.programcreek.com/python/example/5304/socket.has_ipv6 how you can use the socket library.

Reply:
```
def _has_ipv6():                                                                 
    """ Returns True if the system can bind an IPv6 address. """                 
    sock = None                                                                  
    has_ipv6 = False                                                             
                                                                                 
    if socket.has_ipv6:                                                                                     
        try:                                                                     
            sock = socket.socket(socket.AF_INET6)                                
            sock.bind(("::1", 0))                                                
            has_ipv6 = True                                                      
        except Exception:                                                        
            pass                                                                 
                                                                                 
    if sock:                                                                     
        sock.close()                                                             
    return has_ipv6
```
this work, the host is replaced by '::1', 

use socket or netifaces, any ideas?